# One Sample Z Test

Performed when the population variance or standard deviation are known.

## Example-1
- Suppose that a beach is safe to swim if the mean level of lead in the water is 10.0 (μ0) parts/million.
- We assume Xi ~ N(μ, σ = 1.5)
- Water safety is going to be determined by taking 40 water samples and using the test statistic.
- Sample mean = 10.5
- α = 0.05


Bu örnek, bir istatistiksel hipotez testini ve güven aralıklarını anlamaya yönelik bir uygulamadır. Örnekte, bir plajın yüzmek için güvenli olup olmadığını belirlemek amacıyla su örneklerinde kurşun seviyeleri analiz edilmektedir.

Örnek Açıklaması
Varsayılan Durum (Null Hypothesis, H0): Plajdaki suyun kurşun seviyesi ortalaması güvenli sınırlar içinde, yani 
𝜇
0
=
10.0
μ 
0
​
 =10.0 parça/milyon (ppm).
Alternatif Hipotez (Alternative Hypothesis, H1): Plajdaki suyun kurşun seviyesi ortalaması güvenli sınırların dışında, yani 
𝜇
≠
10.0
μ

=10.0 ppm.
Örnekte kullanılan bilgiler:

Kurşun seviyesi 
𝑋
𝑖
X 
i
​
  ~ N(μ, σ = 1.5), yani normal dağılıma sahip ve standart sapması 1.5.
40 su örneği alınacak ve örnek ortalaması (sample mean) 10.5 ppm olarak bulunmuş.
Test istatistiği kullanılacak.
Testin anlamlılık seviyesi 
𝛼
=
0.05
α=0.05.

In [1]:
#Import Libraries
from scipy import stats
import numpy as np

In [2]:
#H0: mu = 10
#H1: mu > 10

    Write the given variables
    
    Bu değerler, bir hipotez testi yaparken kullanılır ve tipik olarak aşağıdaki soruları cevaplamak için kullanılır:

    Örneklem ortalaması, popülasyon ortalamasından anlamlı derecede farklı mı?
    Popülasyonun standart sapması bilindiğinde, örneklem ortalamasının belirli bir güven düzeyinde popülasyon ortalamasından sapıp sapmadığını anlamak.
    
    These values are used when conducting a hypothesis test and are typically used to answer the following questions:

    Is the sample mean significantly different from the population mean?
    Given the population standard deviation, to determine if the sample mean deviates from the population mean at a certain confidence level.

In [3]:
mu = 10 # population mean

xbar = 10.5  # sample mean

n = 40    # sample size

sigma = 1.5   # population standart deviation

alpha = 0.05   # Significance level for %95

    Calculate the test statistic
    
    Bu kod, z-test istatistiğini hesaplamak için kullanılır. Z-test, popülasyon standart sapması bilindiğinde örneklem ortalamasının popülasyon ortalamasından anlamlı derecede farklı olup olmadığını test etmek için kullanılır.
    
    This code is used to calculate the z-test statistic. The z-test is used to test whether the sample mean is significantly different from the population mean when the population standard deviation is known.

    Bu kod çalıştırıldığında, z değeri yaklaşık olarak 2.11 bulunacaktır. Bu değeri, z-tablolarıyla karşılaştırarak örneklem ortalamasının popülasyon ortalamasından anlamlı derecede farklı olup olmadığını test edebilirsiniz. Örneğin, %95 güven düzeyinde (α = 0.05), z değeri yaklaşık olarak ±1.96'dan büyük veya küçükse, null hipotezi reddedebilirsiniz. Bu durumda, z değeri 2.11 olduğu için, 10.5'lik örneklem ortalamasının 10'luk popülasyon ortalamasından anlamlı derecede farklı olduğunu söyleyebiliriz.
    
    When this code is executed, the z value will be approximately 2.11. You can compare this value to z-tables to test if the sample mean is significantly different from the population mean. For example, at the 95% confidence level (α = 0.05), if the z value is approximately greater than ±1.96, you can reject the null hypothesis. In this case, since the z value is 2.11, we can say that the sample mean of 10.5 is significantly different from the population mean of 10.

In [6]:
z = (xbar - mu) / (sigma/np.sqrt(n))
z

2.1081851067789197

    Calculate the p-value
    
    Bu kod, hesaplanan z-değerine karşılık gelen p-değerini hesaplar. Z-testte, p-değeri örneklem ortalamasının popülasyon ortalamasından ne kadar farklı olduğunu belirler. Düşük bir p-değeri, örneklem ortalamasının popülasyon ortalamasından anlamlı derecede farklı olduğunu gösterir.
    
    
    p-değeri: p-değeri genellikle bir eşik değer (genellikle 0.05) ile karşılaştırılır. Eğer p-değeri bu eşik değerden küçükse, null hipotez reddedilir ve örneklem ortalamasının popülasyon ortalamasından anlamlı derecede farklı olduğu sonucuna varılır. Bu durumda, p-değerinin küçük olması, 10.5'lik örneklem ortalamasının 10'luk popülasyon ortalamasından anlamlı derecede farklı olduğunu gösterir.
    
    
    This code calculates the p-value corresponding to the computed z-score. In a z-test, the p-value indicates how different the sample mean is from the population mean. A low p-value suggests that the sample mean is significantly different from the population mean.

    The p-value is usually compared to a threshold value (commonly 0.05). If the p-value is smaller than this threshold, the null hypothesis is rejected, indicating that the sample mean is significantly different from the population mean. In this case, a small p-value shows that the sample mean of 10.5 is significantly different from the population mean of 10.
    
    

In [7]:
p_value = 1 - stats.norm.cdf(z)
p_value

0.017507490509831247

Conclusion

In [8]:
if p_value < alpha:
    print('At {} level of significance, we can reject the null hypothesis in favor of alternative hypothesis.'.format(alpha))
else:
    print('At {} level of significance, we fail to reject the null hypothesis.'.format(alpha))

At 0.05 level of significance, we can reject the null hypothesis in favor of alternative hypothesis.


## Example-2
- A department store manager determines that a new billing system will be cost-effective only if the mean monthly account is more than 170 dollars.

- A random sample of 400 monthly accounts is drawn, for which the sample mean is 178 dollars.

- The accounts are approximately normally distributed with a standard deviation of 65 dollars.

- Can we conclude that the new system will be cost-effective?

    
    - Bir mağaza müdürü, yeni bir fatura sisteminin maliyet etkin olacağını yalnızca aylık ortalama hesabın 170 dolardan fazla olduğunu belirler.

    - Rastgele seçilen 400 aylık hesap için örneklem ortalama 178 dolardır.

    - Hesaplar yaklaşık olarak 65 dolarlık bir standart sapmaya sahip olarak normal dağılır.

    - Yeni sistemin maliyet etkin olacağını sonuçlandırabilir miyiz?

In [9]:
#H0: mu = 170
#H1: mu > 170

In [10]:
mu = 170

x_bar = 178
n = 400

sigma = 65

alpha = 0.05

In [11]:
z = (x_bar - mu) / (sigma/np.sqrt(n))
z

2.4615384615384617

In [12]:
#Standard Error
sigma/np.sqrt(n)

3.25

In [13]:
p_value = 1 - stats.norm.cdf(z)

In [14]:
1 - stats.norm.cdf(178, 170, 3.25)

0.006917128192854505

In [15]:
if p_value < alpha:
    print('At {} level of significance, we can reject the null hypothesis in favor of alternative hypothesis.'.format(alpha))
else:
    print('At {} level of significance, we fail to reject the null hypothesis.'.format(alpha))

At 0.05 level of significance, we can reject the null hypothesis in favor of alternative hypothesis.


# One Sample t Test

## Example-1
- Bon Air ELEM has 1000 students. The principal of the school thinks that the average IQ of students at Bon Air is at least 110. To prove her point, she administers an IQ test to 20 randomly selected students.
- Among the sampled students, the average IQ is 108 with a standard deviation of 10.
- Based on these results, should the principal accept or reject her original hypothesis? α = 0.01

In [16]:
#H0: mu = 110
#H1: mu < 110

In [17]:
mu = 110

xbar = 108
n = 20
s = 10

alpha = 0.01

In [18]:
t = (xbar - mu) / (s / np.sqrt(n))
t

-0.8944271909999159

In [19]:
p_value = stats.t.cdf(t, df=n-1)
p_value

0.1911420676837155

In [22]:
stats.t.ppf(p_value, df=n-1) # t value

-0.8944271909999159

In [24]:
#How to calculate critical value
critic_value = stats.t.ppf(alpha, df=n-1)
critic_value

-2.5394831906222888

In [27]:
# Hipotez testinin sonucunu belirleme
if t < critic_value:
    print("We reject the null hypothesis. The mean IQ is significantly lower than 110.")
else:
    print("We do not reject the null hypothesis. We cannot conclude that the mean IQ is significantly lower than 110.")

We do not reject the null hypothesis. We cannot conclude that the mean IQ is significantly lower than 110.


    Bu testin sonucuna göre:

    t-value: -0.8944271909999159'dur.
    p-value: 0.1911420676837155'dur.
    Critic value: -2.5394831906222888'dir.
    
    t-value, Critic value'dan büyük olduğu için, null hipotezi reddedemeyiz. Bu nedenle, ortalama IQ'nun 110'dan anlamlı derecede düşük olduğunu söyleyemeyiz.
    
    
    Based on the results of this test:

    t-value: -0.8944271909999159
    p-value: 0.1911420676837155
    Critical value: -2.5394831906222888
    
    Since the t-value is greater than the critical value, we do not reject the null hypothesis. Therefore, we cannot conclude that the mean IQ is significantly lower than 110.